### ISSUE TITLE
Feature proposal: `clear` action creator

### ISSUE THREAD

**Issue by timhwang21** [erikras/redux-form#3055]
### Are you submitting a **bug report** or a **feature request**?

Feature request

A common pattern I'm running into for dependent fields (e.g. field B's value depends on field A having a value) is that I want to clear field B when field A changes.

So I do something like:
```javascript
// Assume field B, C, and D are conditional and A having a value
handleFieldAChange(event, value) {
  const { change } = this.props;

  if (!value) { // if field a was emptied
    change('b', null);
    change('c', null);
    change('d', null);
    untouch('b', 'c', 'd');
  }
}
```

`change` doesn't support passing in multiple fields currently. In my case, I know that the value for each changed field will be `null`, so an action creator `clear()` (or some better name) with the same signature as `untouch()` would be very useful:
```javascript
handleFieldAChange(event, value) {
  const { clear } = this.props;

  if (!value) {
    clear('b', 'c', 'd');
  }
// Behavior: args.forEach(arg => { blur(arg); change(arg, null); })
```

*Alternative / additional suggestion:* Change the argument order of `change` to `(form:string, value:any, ...fields:string)`, to bulk change values. The downside is that similar action creators will need to be changed to be consistent with the signature.

Is there a better existing way that I'm missing? If not, would there be interest in this functionality? If so I can take a crack at it over the weekend. Thanks!

**Comment by danielrob** [comment]
Seems sensible to me, I'd say go for it!

**Comment by danielrob** [comment]
Note as well the linked issue.. maybe there are other cases for having bulk setting of formValues.

**Comment by timhwang21** [comment]
Hm, I just saw the issue, so maybe a `changeMany` or something would be more general-purpose. Thanks! (By the way, I'm not sure what the best way to proceed on this is -- I've filed another issue #3295)

**Comment by danielrob** [comment]
`setFormValues` is another name suggestion ;).

**Comment by danielrob** [comment]
Consider also #2224

**Comment by timhwang21** [comment]
@danielrob So I'm running into an issue in the implementation of a bulk change action, and was hoping one of the maintainers could chime in... (cc @erikras)

The issue: the function signature for actions that change multiple fields is `...fields`, taking rest arguments instead of an array for convenience/ nicer syntax, I guess. This means that the list of fields must be the last arguments passed.

However, internally the `change` action also has a functionality for custom `touch` and `persistentSubmitErrors`. The signature of this function is `(form: string, field: string, value: any, touch: boolean, persistentSubmitErrors: boolean)`. 

Any sort of bulk change action can't have `touch` and `persistentSubmitErrors` as the last argument while also accepting spread field names vs. an array. If these two arguments were moved to the front, however, the exported action creator would have two arguments the user shouldn't care about, as `createReduxForm` should be responsible for passing these arguments based on configuration.

Not supplying these arguments at all would make the bulk change action behave subtly different from the change action. Bug reports galore 😱

The bulk change action **could** simply dispatch an array of change actions, which replicates how users are currently using `change()` to set multiple fields to a given value. But IMO the reducer should ideally handle all values at once for better performance (unless the form values object has some immutable-based optimization for deep-setting values? I've seen the `set` and `get` functions but haven't dived into their implementation details).

Thanks!

**Comment by ste93cry** [comment]
@erikras Do you have any ETA on when this feature will be released in a stable version?

**Comment by erikras** [comment]
Fix published in [`v7.2.0`](https://github.com/erikras/redux-form/releases/tag/v7.2.0).

**Comment by MrBlenny** [comment]
It looks like `clearFields` doesn't clear the store value or errors when `destroyOnUnmount:false`. Is this expected?
